impl/src/main/java/org/jboss/weld/bootstrap/Validator.java

```diff
@@ -689,30 +696,79 @@ private void validateEnabledAlternativeClasses(BeanManagerImpl beanManager, Bean
                 if (enabledClass.isAnnotation() || enabledClass.isInterface()) {
                     throw ValidatorLogger.LOG.alternativeBeanClassNotClass(definition);
                 } else {
-                    // check that the class is a bean class of at least one alternative
-                    boolean alternativeBeanFound = false;
-                    for (Bean<?> bean : beansByClass.get(enabledClass)) {
-                        if (bean.isAlternative()) {
-                            alternativeBeanFound = true;
+                    final WeldConfiguration configuration = beanManager.getServices().get(WeldConfiguration.class);
+                    boolean allowExcludedAlternatives = configuration.getBooleanProperty(ConfigurationKey.ALLOW_EXCLUDED_ALTERNATIVES);
+                    if (! allowExcludedAlternatives) {
+                        // check that the class is a bean class of at least one alternative
+                        boolean alternativeBeanFound = false;
+                        for (Bean<?> bean : beansByClass.get(enabledClass)) {
+                            if (bean.isAlternative()) {
+                                alternativeBeanFound = true;
+                            }
+                        }
```

Comment:
For this for cycle you can use your newly created method `isAlternativeBean` as well right?